xla/pjrt/c/pjrt_c_api_gpu_internal.cc

```diff
@@ -167,12 +167,100 @@ PJRT_Error* PJRT_ExecuteContext_Create(PJRT_ExecuteContext_Create_Args* args) {
   return nullptr;
 }
 
+namespace {
+
+absl::StatusOr<std::unique_ptr<xla::StreamExecutorGpuTopologyDescription>>
+TryParseTopology(
+    xla::PjRtPlatformId platform_id, std::string platform_name,
+    absl::flat_hash_map<std::string, xla::PjRtValueType>& options) {
+  auto topology_it = options.find("topology");
+  if (topology_it == options.end()) {
+    return {nullptr};
+  }
+
+  std::string topology_string = std::get<std::string>(topology_it->second);
+
+  int num_slices;
+  int num_hosts_per_slice;
+  int num_devices_per_host;
+  std::vector<std::string> topology_components =
+      absl::StrSplit(topology_string, 'x');
+  if (topology_components.size() != 3 ||
+      !absl::SimpleAtoi(topology_components[0], &num_slices) ||
+      !absl::SimpleAtoi(topology_components[1], &num_hosts_per_slice) ||
+      !absl::SimpleAtoi(topology_components[2], &num_devices_per_host)) {
+    return absl::InternalError(
+        "topology must be of shape "
+        "\"<num-slices>x<num-hosts-per-slice>x<num-devices-per-host>\"");
+  }
+
+  std::string target_config_proto_string;
+  const auto& debug_options = xla::GetDebugOptionsFromFlags();
+
+  auto target_config_it = options.find("target_config");
```

Comment:
TBH this part never made sense to me, if we already have a debug flag/a separate option to pass through debug_config, do we need to put it into `options` as well?

Reply:
So I am not a fan of the debug flag solution. My reasons:

* The name **debug** flag suggests it's for debugging. `target_config` is not for debugging.
* Global flag does not let you create a compile server that can compile for different GPUs easily.
* The `xla_gpu_target_config_filename` flag forces you to create a file for `target_config` and to use the filesystem. This is not always convenient.
* Something as important as target_config should be an explicit call parameter rather than a global flag.

Am I missing something?